### PR TITLE
[macOS] Add retry logic

### PR DESCRIPTION
**Pull request by vsafonkin**
# Description
Added retry logic to download some tools:
- aws
- azcopy
- node
- openjdk
- pypy
- stack
- xamarin

Added retry ps helper for toolsets.

#### Related issue:
internal issue: [#1031](https://github.com/actions/virtual-environments-internal/issues/1031)

## Check list
- [x] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/macos/provision/core/aws.sh [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #!/bin/bash -e -o pipefail
 
+source ~/utils/utils.sh
+
 echo Installing aws...
-curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
+AWSCLIURL="https://awscli.amazonaws.com/AWSCLIV2.pkg"
+download_with_retries $AWSCLIURL "."
```
Comment (maxim-lobanov) [inline images/macos/provision/core/aws.sh:7]: Is it better to download to `/tmp` like you have done in other places?

File: images/macos/provision/core/azcopy.sh [hunk 2]
```diff
@@ -1,8 +1,10 @@
 #!/bin/bash -e -o pipefail
 
+source ~/utils/utils.sh
+
 AZCOPY_DOWNLOAD_URL="https://aka.ms/downloadazcopy-v10-mac"
 
-wget -O "$HOME/azcopy.zip" "$AZCOPY_DOWNLOAD_URL"
+download_with_retries $AZCOPY_DOWNLOAD_URL "." "azcopy.zip"
```
Comment (maxim-lobanov) [inline images/macos/provision/core/azcopy.sh:7]: Is it better to download to `/tmp` like you have done in other places?

File: images/macos/provision/core/aws.sh (hunk 1 above)
Comment (vsafonkin) [reply images/macos/provision/core/aws.sh]: Done

File: images/macos/provision/core/azcopy.sh (hunk 2 above)
Comment (vsafonkin) [reply images/macos/provision/core/azcopy.sh]: Done

File: images/macos/provision/core/aws.sh [hunk 3]
```diff
@@ -1,9 +1,13 @@
 #!/bin/bash -e -o pipefail
 
+source ~/utils/utils.sh
+
 echo Installing aws...
-curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
+AWSCLIURL="https://awscli.amazonaws.com/AWSCLIV2.pkg"
+download_with_retries $AWSCLIURL "/tmp"
+pushd /tmp
```
Comment (maxim-lobanov) [inline images/macos/provision/core/aws.sh:8]: Can we just do `sudo installer -pkg /tmp/AWSCLIV2.pkg -target /` and get rid of pushd and popd?

**Review by maxim-lobanov — APPROVED** [review]

File: images/macos/provision/core/aws.sh [hunk 4]
```diff
@@ -1,9 +1,13 @@
 #!/bin/bash -e -o pipefail
 
+source ~/utils/utils.sh
+
 echo Installing aws...
-curl "https://awscli.amazonaws.com/AWSCLIV2.pkg" -o "AWSCLIV2.pkg"
+AWSCLIURL="https://awscli.amazonaws.com/AWSCLIV2.pkg"
```
Comment (AlenaSviridenko) [inline images/macos/provision/core/aws.sh:6]: could you please add underscores to variable name? like AWS_CLI_URL to make it more readable

**Review by AlenaSviridenko — APPROVED** [review]

File: images/macos/provision/core/aws.sh (hunk 4 above)
Comment (vsafonkin) [reply images/macos/provision/core/aws.sh]: Done

**Review by al-cheb — APPROVED** [review]

File: images/macos/provision/core/aws.sh (hunk 3 above)
Comment (miketimofeev) [reply images/macos/provision/core/aws.sh:8]: And I'm not sure that we prune /tmp at the end of the build, so I'd suggest clean it up manually

Comment (vsafonkin) [reply images/macos/provision/core/aws.sh]: Done

**Review by miketimofeev — APPROVED** [review]

**Pull request merged**